### DISCUSSION TITLE
Multiple pages by bundle

### DISCUSSION THREAD

**Discussion by arcanis** [vercel/next.js#18002] [Help]
I have an application where multiple very similar pages have different URLs that the user is expected to all browse during a single session (in a sense, the URLs are really just a serialized state). Because of this, it seems like Next is generating one bundle for each entry point, leading to an annoying loading experience (navigating from one url to another is delayed by the chunk being loaded).

Is there a way to tell Next that some pages should go inside the same bundle?

**Comment by timneutkens** [comment]
If you want to route certain patterns to a single page you can use the `rewrites` feature: https://nextjs.org/docs/api-reference/next.config.js/rewrites

**Reply by lucasbasquerotto** [reply]
@timneutkens The rewrites will change the url the user is accessing (like a redirect) or just change the component used? (so that the page used will be the page provided in the destination, but the url will be the same as it was previously, making it transparent to the user)

**Reply by timneutkens** [reply]
`rewrites` are rewrites that allow you to map a certain url to a page. `redirects` is another feature and that handles redirecting a page to a completely different url

**Reply by arcanis** [reply]
@timneutkens Let's imagine I have `/[paramA]` and `/[paramA]/[paramB]`; to put them in the same bundle, I'd need to rewrite like this:

```json
{ "source": "/[paramA]", "destination": "/[paramA]/ignorethisparam" }
```

Then move the code from the source route into the destination route, and branch so that detecting `ignorethisparam` as `paramB` would instead render the source route component - in essence, I'd have a single page, not two. Is that correct?

I wonder if there could be a simpler approach to define which pages are bundled together ... it feels a bit uncomfortable to bend arguments like this. At this rate, I could as well use a catch-all route and simply make the routing myself 🤔

**Comment by karanpvyas** [comment]
Use Component Composition properly and most of your code would already be loaded inside the browser when your first page is loaded.

Successive loads would+should be very fast after that.


OR

Have a single page handle multiple urls via multiple route params.

pages/[dirRouteParam1]/[dirRouteParam2]/[routeParam3].tsx
or [...allParams].tsx ofc.
